project2/libs/tests/test_utils.h

```diff
@@ -3,19 +3,33 @@
 
 #include "uart.h"
 
-#define Assert(expr)                         \
-    {                                        \
-        if ((!(expr)) || PORTE != 0) {       \
-            UART_print(                      \
-                "Assert Failed! %s : %d\n",  \
-                __FILE__, __LINE__);         \
-            for (;;) {}                      \
-        }                                    \
-    }
+#define Assert(expr)                     \
+{                                        \
+    if (!(expr)) {                       \
+        UART_print(                      \
+            "Assert Failed! "            \
+            "%s : %d\n",                 \
+            __FILE__, __LINE__);         \
+        for (;;) {}                      \
+    }                                    \
+}
 
-#define AssertNeverCalled()                     \
-    {                                           \
-        PORTE = 0xFF;                           \
-    }
+#define AssertAborted()                  \
+{                                        \
```

Comment:
This isn't the best way of doing things yet. I want to have `AssertAborted` and `AssertNeverCalled` independent. Right now if never called was called, then an AssertAborted might reset that

Reply:
Low half of PORTE is for aborts, high half is for never called. Kinda silly still, but at least they don't interfere anymore